SalesforceCommerceCloud/pwa-kit
# @W-16102728@ Clear other sessions on password change

Prior to this change, when the shopper changes their passwords, all existing SCAPI sessions are invalidated by the API.

However, if you have sessions active on other devices or browsers, the invalidated access token will remain until it expires by age. As a result, other browsers and devices get stuck in an invalid state until the access token clears out.

This PR updates our useQuery and useMutation hooks to introduce some error handling to clear the auth state if it has been found that a SCAPI request failed because of an invalidated session.

Test steps:

- Open 2 sessions (can use 1 non-incognito and 1 incognito window)
- On both sessions, log in to the same account
- On one of the sessions, change your password. Verify that for this session you remain logged in after changing your password.
- On the other session, do something that will make a SCAPI call (ie. try to pull up a category). Verify that you are logged out and taken to the log in screen.
